typescript.js

```diff
@@ -0,0 +1,68 @@
+module.exports = {
+    extends: './index',
+    parser: '@typescript-eslint/parser',
+    rules: {
```

Comment:
That is a great but complicated question with a twisted answer.

TL;DR No, as with @typescript-eslint/recommended we want to [support unknown file types](https://github.com/typescript-eslint/typescript-eslint/pull/741#issuecomment-552694613). Overrides are, ["applied only if the glob patterns of both of the parent config and the child config [match],"](https://eslint.org/docs/user-guide/configuring#how-it-works) and would restrict usage to the defined extensions.

The @typescript-eslint/eslint-recommended ruleset is, ["meant to be used after extending eslint:recommended."](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/configs#eslint-recommended) They are not extension rules, and I think the override is used to ensure they are only applied where the TypeScript compiler would be used, e.g. .ts files but not markdown, etc.

There is also the question of if the TypeScript configuration should ever be run on JavaScript files. [There are TypeScript rules that are not compatible with JavaScript, and @typescript-eslint recommends using overrides.](https://github.com/typescript-eslint/typescript-eslint/issues/109#issuecomment-530909615) There are also performance issues when using rules that require type checking

Reply:
That is a great but complicated question with a twisted answer.

TL;DR No, as with @typescript-eslint/recommended we want to [support unknown file types](https://github.com/typescript-eslint/typescript-eslint/pull/741#issuecomment-552694613). Overrides are, ["applied only if the glob patterns of both of the parent config and the child config [match],"](https://eslint.org/docs/user-guide/configuring#how-it-works) and would restrict usage to the defined extensions.

The @typescript-eslint/eslint-recommended ruleset is, ["meant to be used after extending eslint:recommended."](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/configs#eslint-recommended) They are not extension rules, and I think the override is used to ensure they are only applied where the TypeScript compiler would be used, e.g. .ts files but not markdown, etc.

There is then the question of if the TypeScript configuration should ever be run on JavaScript files. [There are a few TypeScript rules that are not compatible with JavaScript, and @typescript-eslint recommends using overrides.](https://github.com/typescript-eslint/typescript-eslint/issues/109#issuecomment-530909615) It may be reasonable to advise that the TypeScript configuration can be run on JavaScript if there aren't many problematic rules in our final configuration. However, if the workspace has many JavaScript files that are not included in the compiler configuration then they may require an override to avoid [performance problems](https://github.com/angular-eslint/angular-eslint#eslint-configs-and-performance).

At the least, I think it's reasonable to include an explanation and example of linting a mixed workspace with overrides. We can make a recommendation on when to use each approach once we have a configuration.

Reply:
I see, that makes sense. I was just curious because I saw the usage of overrides at eslint-recommended. I agree that we should focus on the TypeScript rules for now and probably include some documentation at the end for linting a mixed workspace. Thanks for the explanation!